src/extension.ts

```diff
@@ -61,38 +61,51 @@ class RunFormatOnSave {
 			const overReactFormatRangeIsValid = dependencyHasValidMinVersion(this.overReactFormatKey, this.minOverReactFormatVersion, readFileSync(`${this.projectDir}/pubspec.yaml`, 'utf-8'), true);
 
 			if (pubspecContainsOverReactFormat && !overReactFormatRangeIsValid) {
-				this.showChannelMessage(`OverReact Format range is not acceptable. Bump the minimum to 3.1.0 to use OverReact Format on Save.`);
+				this.showChannelMessage('over_react_format range is not compatible with OverReact Format on Save.'
+				+ ' Bump the minimum to 3.1.0 to use OverReact Format on Save. Defaulting to using dartfmt instead.');
 			}
 			this.useOverReactFormat =  pubspecContainsOverReactFormat && overReactFormatRangeIsValid;
 		}
 		// No else condition because there's no penalty for the project not being a Dart project.
 		// The `onDocumentSave` command will just be short-circuited if it is run on non-Dart files.
 	}
 
-	buildCommand(fileName: string) {
-		let command:string;
-
+	startProcess(fileName: string) : process.ChildProcess {
 		const customLineLength = this.config.get<Number>('customLineLength', 0);
 		const shouldDetectLineLength = this.config.get<Boolean>('detectCustomLineLength');
 		const shouldUseCustomLineLength = customLineLength > 0;
+		let executable : "pub" | "dartfmt";
```

Comment:
whoa I don't know TS well - does this throw a type error if you try to set executable to anything other than "pub" or "dartfmt"?

Reply:
Yeah! Here's the doc reference: typescriptlang.org/docs/handbook/release-notes/typescript-1-8.html#string-literal-types

One of my favorite parts of TypeScript is how it can handle union types or weird mixtures of types